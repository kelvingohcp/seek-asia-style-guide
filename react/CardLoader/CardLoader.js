import React from 'react';
import Card from '../Card/Card';
import styles from './CardLoader.less';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const CardLoader = ({ children, className }) => {
  return (
    <Card className={classnames(styles.root, className)}>
      <div className={styles.animatedBackground}>
        {children}
      </div>
    </Card>
  );
};

CardLoader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

export default CardLoader;
