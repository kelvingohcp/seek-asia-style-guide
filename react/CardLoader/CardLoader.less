@import (reference) "~seek-asia-style-guide/theme";

.root {
  background: @saWhite;
  width: 100%;
  padding: @grid-gutter-width;
}

.animatedBackground {
  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeHolderShimmer;
  animation-timing-function: linear;
  background: @sk-gray-lightest;
  background: linear-gradient(75deg, @sk-gray-light 8%, @sk-mid-gray-light 18%, @sk-gray-light 33%);
  background-size: 200% 100%;
  position: relative;
}

@keyframes placeHolderShimmer {
  0% {
    background-position: 95% 0%;
  }
  100% {
    background-position: -95% 0%;
  }
}
