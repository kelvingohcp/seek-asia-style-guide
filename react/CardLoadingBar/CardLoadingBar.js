import React from 'react';
import PropTypes from 'prop-types';
import styles from './CardLoadingBar.less';
import classnames from 'classnames';
import { getMultipliedGridLength } from '../../theme/layout/grid';

const CardLoadingBar = ({ width = 0, height = 1, inverse = false, className }) => {
  const HEIGHT_MULTIPLIER = 3;
  const barHeight = getMultipliedGridLength(height * HEIGHT_MULTIPLIER);
  const barWidth = Number.isInteger(width) ? getMultipliedGridLength(width) : width;
  return (
    <div className={classnames(styles.loadingBar, className)} style={{ height: barHeight }}>
      <div
        className={styles.maskedLoadingBar}
        style={{ [(inverse) ? 'right' : 'left']: barWidth }}
      />
    </div>
  );
};

CardLoadingBar.propTypes = {
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.number,
  inverse: PropTypes.bool,
  className: PropTypes.string
};

export default CardLoadingBar;
