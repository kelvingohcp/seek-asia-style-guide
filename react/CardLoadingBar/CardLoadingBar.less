@import (reference) "~seek-asia-style-guide/theme";

.loadingBar {
  position: relative;
}

.maskedLoadingBar {
  left: 0;
  right: 0;
  height: inherit;
  background: @saWhite;
  position: absolute;
}
