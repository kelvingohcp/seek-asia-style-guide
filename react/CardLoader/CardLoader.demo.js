import React from 'react';
import PropTypes from 'prop-types';
import CardLoader from './CardLoader';
import CardLoadingBar from '../CardLoadingBar/CardLoadingBar';
import PageBlock from '../PageBlock/PageBlock';

const cardLoadingBars = [
  (<div style={{ display: 'flex' }} key={1}>
    <div style={{ width: '30%' }}>
      <CardLoadingBar width={'80%'} height={5} />
    </div>
    <div style={{ width: '35%' }}>
      <CardLoadingBar width={40} />
      <CardLoadingBar />
      <CardLoadingBar width={'30%'} />
      <CardLoadingBar />
      <CardLoadingBar width={'30%'} />
    </div>
    <div style={{ width: '35%' }}>
      <CardLoadingBar width={40} inverse />
      <CardLoadingBar />
      <CardLoadingBar width={'30%'} inverse />
      <CardLoadingBar />
      <CardLoadingBar width={20} inverse />
    </div>
  </div>),
  (<CardLoadingBar key={2} />),
  (<CardLoadingBar width={'100%'} key={3} />)
];

const CardLoaderContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '100%' }}>
      <DemoComponent {...componentProps} >
        {cardLoadingBars}
      </DemoComponent>
    </PageBlock>
  );
};

CardLoaderContainer.propTypes = {
  component: PropTypes.element,
  componentProps: PropTypes.object
};

export default {
  route: '/CardLoader',
  title: 'Card Loader',
  component: CardLoader,
  container: CardLoaderContainer,
  initialProps: {
    children: cardLoadingBars
  },
  options: []
};
