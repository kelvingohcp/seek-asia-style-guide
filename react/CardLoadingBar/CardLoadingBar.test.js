import React from 'react';
import { shallow } from 'enzyme';

import CardLoadingBar from './CardLoadingBar';
import CardLoader from '../CardLoader/CardLoader';

describe('CardLoadingBar', () => {
  it('should render with empty loadingBar', () => {
    expect(shallow(
      <CardLoader >
        <CardLoadingBar />
      </CardLoader>
    )).toMatchSnapshot();
  });

  it('should render with full width loadingBar', () => {
    expect(shallow(
      <CardLoader >
        <CardLoadingBar width={'1oo%'} />
      </CardLoader>
    )).toMatchSnapshot();
  });

  it('should render loadingBar width and height in px', () => {
    expect(shallow(
      <CardLoader >
        <CardLoadingBar width={10} height={4} />
      </CardLoader>
    )).toMatchSnapshot();
  });

  it('should render loadingBar height in px', () => {
    expect(shallow(
      <CardLoader >
        <CardLoadingBar width={10} height={4} />
      </CardLoader>
    )).toMatchSnapshot();
  });

  it('should render loadingBar from right', () => {
    expect(shallow(
      <CardLoader >
        <CardLoadingBar width={'5o%'} inverse />
      </CardLoader>
    )).toMatchSnapshot();
  });
});
