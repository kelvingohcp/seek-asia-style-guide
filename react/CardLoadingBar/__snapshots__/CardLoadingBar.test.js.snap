// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLoadingBar should render loadingBar from right 1`] = `
<Card
  className="root"
>
  <div
    className="animatedBackground"
  >
    <CardLoadingBar
      inverse={true}
      width="5o%"
    />
  </div>
</Card>
`;

exports[`CardLoadingBar should render loadingBar height in px 1`] = `
<Card
  className="root"
>
  <div
    className="animatedBackground"
  >
    <CardLoadingBar
      height={4}
      width={10}
    />
  </div>
</Card>
`;

exports[`CardLoadingBar should render loadingBar width and height in px 1`] = `
<Card
  className="root"
>
  <div
    className="animatedBackground"
  >
    <CardLoadingBar
      height={4}
      width={10}
    />
  </div>
</Card>
`;

exports[`CardLoadingBar should render with empty loadingBar 1`] = `
<Card
  className="root"
>
  <div
    className="animatedBackground"
  >
    <CardLoadingBar />
  </div>
</Card>
`;

exports[`CardLoadingBar should render with full width loadingBar 1`] = `
<Card
  className="root"
>
  <div
    className="animatedBackground"
  >
    <CardLoadingBar
      width="1oo%"
    />
  </div>
</Card>
`;
